src/ansiblelint/utils.py

```diff
@@ -901,3 +898,22 @@ def nested_items(
             yield "list-item", item, parent
             for k, v, p in nested_items(item):
                 yield k, v, p
+
+
+def nested_items_path(
+    data: Union[Dict[Any, Any], List[Any]],
+    parent_path: Optional[List[Union[str, int]]] = None,
+) -> Iterator[Tuple[Any, Any, List[Union[str, int]]]]:
+    """Iterate a nested data structure."""
+    if parent_path is None:
+        parent_path = []
+    if isinstance(data, dict):
+        for key, value in data.items():
+            yield key, value, parent_path
+            for k, v, p in nested_items_path(value, parent_path + [key]):
+                yield k, v, p
+    if isinstance(data, list):
+        for index, item in enumerate(data):
+            yield index, item, parent_path
+            for k, v, p in nested_items_path(item, parent_path + [index]):
+                yield k, v, p
```

Comment:
Looking at this function more, I see a huge opportunity to get rid of the duplication and some nesting:
```suggestion

    if isinstance(data, dict):
        convert_to_tuple = dict.items
    elif isinstance(data, list):
        convert_to_tuple = enumerate
    else:
        raise TypeError

    for key, value in convert_to_tuple(data):
        yield key, value, parent_path
        yield from nested_items_path(value, parent_path + [key])
```
The only thing I don't understand is that why the second loop inverts key and value — I suppose it's a bug.

Reply:
Nice! Yeah, I'll adjust it like this.

Reply:
done.

Reply:
@cognifloyd could you confirm that the swapped arguments in the last recursive call were indeed a bug? Because I was reviewing from my phone and might've missed something. Just want to be sure that this guess was correct.

Reply:
Ah, never mind. I see you replied at https://github.com/ansible-community/ansible-lint/pull/1824#pullrequestreview-860164964.